src/utils/slot.py

```diff
@@ -90,25 +83,49 @@ def get_first_non_missed_slot(
     if not existing_header:
         raise NoSlotsAvailable('No slots available for current report. Check your CL node.')
 
-    if ref_slot_is_missing:
-        # Ref slot is missing. We need to check the parent root found non-missing slot
-        non_missed_header_parent_root = existing_header.data.header.message.parent_root
-        parent_header = cc.get_block_header(non_missed_header_parent_root)
+    _check_block_header(existing_header)
+    return slot_is_missing, existing_header
 
-        if int(parent_header.data.header.message.slot) >= slot:
-            raise InconsistentData(
-                "Parent root next to the ref slot's existing header doesn't match the expected slot.\n"
-                'Probably, a problem with the consensus node.'
-            )
 
-        if direction == 'back':
-            _check_block_header(existing_header)
-            existing_header = parent_header
+def get_prev_non_missed_slot(
+    cc: ConsensusClient,
+    slot: SlotNumber,
+    last_finalized_slot_number: SlotNumber,
+) -> BlockDetailsResponse:
+    """
+    Get non-missed slot data. In case of missed slot, we take parent root and get parent slot data.
+    """
+    slot_is_missing, existing_header = _get_closest_non_missed_headers(cc, slot, last_finalized_slot_number)
 
-    _check_block_header(existing_header)
-    slot_details = cc.get_block_details(existing_header.data.root)
-    logger.info({'msg': f'Resolved to slot: {slot_details.message.slot}'})
-    return slot_details
+    if not slot_is_missing:
+        return cc.get_block_details(existing_header.data.root)
+
+    non_missed_header_parent_root = existing_header.data.header.message.parent_root
+    parent_header = cc.get_block_header(non_missed_header_parent_root)
+
+    if (
+        int(parent_header.data.header.message.slot) >= slot
+        or int(existing_header.data.header.message.slot) - int(parent_header.data.header.message.slot) < 1
```

Comment:
in another words:
existing_header.data.header.message.slot == parent_header.data.header.message.slot

Reply:
possibly we can check that
block_number between existing and parent is 1

Reply:
for some reason existing_header slot can be less than parent_header slot from node perspective, so with the current solution we will catch this